### PR TITLE
Implement labelling w/ updated functionality

### LINKED ISSUE #64: [FEATURE] Create utility function to apply {labelled} labels to supertibble
# Feature Request Description
A new exported function (`make_labelled()`) that uses the [`labelled`](http://larmarange.github.io/labelled/) package to apply labels to fields in the supertibble and fields within the supertibble's list columns.

# Proposed Solution
A use would run:
```r
suppertibble <- make_labelled(suppertibble)
```
which would:

1. Apply labels to the variables in the supertibble. These will be defined by `REDCapTidieR`.
2. Apply labels to the variables in `supertibble$redcap_metadata` and `supertibble$redcap_events`. These will be defined by `REDCapTidieR`.
3. Apply labels to the variables in `supertibble$redcap_data`. These would be derived from the contents of `supertibble$redcap_metadata`

Additionally we should add functionality for editing the labels in `redcap_metadata` before applying them to `redcap_data`. Two potential options:
1. Provide additional parameters to `make_labelled()` that control custom formatting, ex. `remove_terminal_colons`
2. Add a parameter that let's the user pass a formatting function to preprocess labels and give a sensible default, ex:
```r
formatter <- function(x) stringr::str_remove(x, ":$")

suppertibble <- make_labelled(suppertibble, label_format = formatter)
```

Finally, rather than importing `labelled` we should check that it's installed in `make_labelled()` and issue an error message if it isn't that asks the user to install it.

### Checklist

- [x] The issue is atomic
- [x] The issue description is documented
- [x] The issue title describes the problem succinctly
- [x] Developers are assigned to the issue
- [x] Labels are assigned to the issue

### PR DESCRIPTION
**Pull request by ezraporter**
# Description
This PR combines the work discussed in #65 to implement the `make_labelled()` function to apply `labelled` labels. The functionality, documentation, and tests are complete and vignettes will be updated in a separate PR.

The basic functionality is:

```r
supertbl <- tibble::tribble(
    ~ redcap_data, ~ redcap_metadata,
    tibble(x = letters[1:3]), tibble(field_name = "x", field_label = "X Label:")
  )

# Use the labels as is (default behavior):
make_labelled(supertbl)
make_labelled(supertbl, format_labels = NULL)

# Supply a function:
make_labelled(supertbl, format_labels = fmt_strip_trailing_colon)

# Supply multiple functions to apply in order:
make_labelled(supertbl, format_labels = c(fmt_strip_trailing_colon, tolower))

# Also works with quoted names of functions or formula functions:
make_labelled(supertbl, format_labels = c("tolower", ~gsub(":", "!", .)))
```

# Proposed Changes 
List changes below in bullet format:

- Add `make_labelled()` function to apply labels to columns of:
    - supertibble (pre-defined)
    - elements of `supertibble$redcap_metadata` (pre-defined based on REDCap data dictionary csv)
    - elements of `supertibble$redcap_data` (based on `supertibble$redcap_metadata`)
    - elements of `supertibble$redcap_events` (pre-defined)
- Add internal `resolve_formatter()` helper function to turn handle `format_labels` argument and return a function that can be applied to labels
- Add exported formatting helpers which are all thin wrappers around `stringr` functions
- Update test cases and pkgdown

### Issue Addressed
Closes #64

### PR Checklist
Before submitting this PR, please check and verify below that the submission meets the below criteria:

- [x] New/revised functions have associated tests
- [x] New/revised functions that update downstream outputs have associated static testing files (`.RDS`) updated under `inst/testdata/create_test_data.R`
- [x] New tests that make API calls use `httptest::with_mock_api` and any new mocks were added to `tests/testthat/fixtures/create_httptest_mocks.R`
- [x] New/revised functions use appropriate naming conventions
- [x] New/revised functions don't repeat code
- [ ] Code changes are less than **250** lines total
- [x] Issues linked to the PR using [GitHub's list of keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] The appropriate reviewer is assigned to the PR
- [x] The appropriate developers are assigned to the PR

# Code Review
This section to be used by the reviewer and developers during Code Review after PR submission

### Code Review Checklist

- [ ] I checked that new files follow naming conventions and are in the right place
- [ ] I checked that documentation is complete, clear, and without typos
- [ ] I added/edited comments to explain "why" not "how"
- [ ] I checked that all new variable and function names follow naming conventions
- [ ] I checked that new tests have been written for key business logic and/or bugs that this PR fixes
- [ ] I checked that new tests address important edge cases

### CODE REVIEW

File: tests/testthat/test-labelled.R [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    redcap_event = "Event Name",
+    redcap_arm = "Arm Name",
+    arm_name = "Arm Description"
+  )
+
+  expect_equal(event_labs, expected_event_labs)
+
+})
+
+test_that("make_labelled handles supertibble with extra columns", {
```
Comment (rsh52) [inline tests/testthat/test-labelled.R:173]: Nice, I like that you tested for this 👍

**Review by rsh52 — APPROVED** [review]
I think this all looks great, and the test suite is very robust. I can't think of any other edge cases right now, so all good from my end! 🎉
